File: 179. Largest Number/Python/Solution.py
class Solution:
    # @param {integer[]} nums
    # @return {string}
    def largestNumber(self, nums):
        strs = [str(num) for num in nums]
        strs.sort(lambda x, y: 1 if x + y < y + x else -1 if x + y > y + x else 0)
        if strs[0] == '0':
            return '0'
        return ''.join(strs)

File: 179. Largest Number/C++/Solution.cpp
class Solution {
public:
    string largestNumber(vector<int> &nums) {
        vector<string> strs(nums.size());
        for (int i = 0; i < strs.size(); i++) {
            strs[i] = to_string(nums[i]);
        }
        sort(strs.begin(), strs.end(), [](string & a, string & b) {
            return a + b > b + a;
        });
        if (strs[0] == "0") {
            return "0";
        }
        string result = "";
        for (string str : strs) {
            result += str;
        }
        return result;
    }
};